### PR TITLE
GH Actions: added ldap and geoip-mmdb tests. Removed from CircleCI.

### PR DESCRIPTION
**Pull request by romeroalx**
### Short description
Added LDAP and geoip-mmdb tests to GitHub Actions. Removed from CircleCI. Simplified collector job in build-and-test-all.yml.

### Checklist

I have:
- [ ] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [ ] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

**Review by fredmorcos — APPROVED** [review]
It seems fine to me. But as with all things CI please wait for a review from @Habbie

File: .github/workflows/build-and-test-all.yml
Comment (Habbie) [inline .github/workflows/build-and-test-all.yml:491]: If we ever add a job, and forget to add it to `needs`, you're likely to hit that job as "not yet succeeded" here I think. So I think the diff (or something like it) might still be necessary?

File: .github/workflows/build-and-test-all.yml
Comment (Habbie) [inline .github/workflows/build-and-test-all.yml:300]: nice cleanup!

**Review by Habbie — CHANGES_REQUESTED** [review]
ldap+geoip tests look great. left one comment on the `collect` thing.

File: .github/workflows/build-and-test-all.yml
Comment (romeroalx) [reply .github/workflows/build-and-test-all.yml:491]: Sure, we could add a third step for checking other jobs not included in the needs section. However, bear in mind that if that job has not been included in the needs section, it could (potentially) still be running when the check for other jobs is executed.

Comment (Habbie) [reply .github/workflows/build-and-test-all.yml:491]: Right, I see my first comment was incorrect. Your check looks at `needs` so it would never run into the value. Then indeed we need something that checks whether `needs` is still complete.

Comment (romeroalx) [reply .github/workflows/build-and-test-all.yml:491]: Added check to fail job also if the list of prerequired jobs is not complete (new jobs added and not reflected there).

**Review by Habbie — APPROVED** [review]

**Pull request merged**
